From ef34e74b67128041a8cd7c0c7efba10a2a31d6ea Mon Sep 17 00:00:00 2001
From: Wolfgang Grandegger <wg@aries-embedded.de>
Date: Sat, 30 Oct 2021 19:41:40 +0200
Subject: [PATCH 2/3] usb: musb: start a new session also if vbus error occurs
 with state a_idle

On the M100PFSEVP board, USB hotplug does not work otherwise.

Signed-off-by: Wolfgang Grandegger <wg@aries-embedded.de>
---
 drivers/usb/musb/musb_core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/musb/musb_core.c b/drivers/usb/musb/musb_core.c
index fc0457db62e1..f5757fdf6a3f 100644
--- a/drivers/usb/musb/musb_core.c
+++ b/drivers/usb/musb/musb_core.c
@@ -786,6 +786,7 @@ static void musb_handle_intr_vbuserr(struct musb *musb, u8 devctl)
 		 */
 	case OTG_STATE_A_WAIT_BCON:
 	case OTG_STATE_A_WAIT_VRISE:
+	case OTG_STATE_A_IDLE:
 		if (musb->vbuserr_retry) {
 			void __iomem *mbase = musb->mregs;
 
-- 
2.28.0

